client/my-sites/current-site/index.jsx

```diff
@@ -95,11 +95,7 @@ module.exports = React.createClass( {
 			return null;
 		}
 		const { hostname } = url.parse( site.URL );
-		let href = 'https://wordpress.com/my-domains';
-
-		if ( config.isEnabled( 'upgrades/domain-management/list' ) ) {
-			href = paths.domainManagementList( site.domain );
-		}
+		let href = paths.domainManagementList( site.domain );
```

Comment:
I think we could even remove this variable and inline this code inside `<NoticeAction />` below.